pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -118,6 +119,36 @@ func (ngm *nvidiaGPUManager) Capacity() v1.ResourceList {
 	}
 }
 
+func containsInt(s string) bool {
+	for _, c := range s {
+		if unicode.IsDigit(c) {
+			return true
+		}
+	}
+	return false
+}
+
+// fixed #42412, release the GPU resources
+func (ngm *nvidiaGPUManager) UpdateDevices(podUID string, exitedContainerID string) {
+	inspectJSON, err := ngm.dockerClient.InspectContainer(exitedContainerID)
+	if err != nil {
+		glog.Errorf("SyncPod Inspect JSON container:%v, error:%v", exitedContainerID, err)
```

Comment:
I think this method should return the error, and handle it in the caller instead. Also change the text to `failed to inspect exited container %q: %v`